src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -510,9 +612,14 @@ public static FormulaValue Mod(IRContext irContext, NumberValue[] args)
             var arg1 = args[1].Value;
 
             // r = a – N × floor(a/b)
-            var q = (int)Math.Floor(arg0 / arg1);
+            var q = (long)Math.Floor(arg0 / arg1);
             var result = arg0 - (arg1 * q);
 
+            if (result < -Math.Abs(arg1) || result > Math.Abs(arg1))
+            {
+                result = 0;
+            }
```

Comment:
Why this condition here? Can you add a comment?

Reply:
When we have extremely large number (1e308) the calculated reminder could be completely wrong,hence this extra validation

I can add a comment if you want to explain this